### PR TITLE
Error message rework 5 - origin or reason unknown

### PR DESCRIPTION
**Pull request by fsimonis**
This PR revises all errors in section 5 **except** the ones deep inside the cplschemes in order not to conflict with #689.

Examples:
```
// getMeshVertexIDsFromPositions
ERROR:  Unable to find a vertex on mesh "MeshOne" at position (0, 1). The request failed for query 2 out of 3.
```

Related to #698

### CODE REVIEW

**Comment by fsimonis** [comment]
That's actually quite interesting.

In commit 893b6a0be694eb86c9bb7e3ee01c63d0d0a52e32, I pulled a check from `void ScaleByAreaAction::performAction(double fullDt)` up to the configuration.
The configuration tests now fail as they were implemented for 3D meshes, which is actually a wrong use-case.

This should probably be a lesson to us that failing early is good. :wink:

File: src/cplscheme/BaseCouplingScheme.cpp [hunk 1]
```diff
@@ -261,7 +261,7 @@ void BaseCouplingScheme::extrapolateData(DataMap &data)
       utils::shiftSetFirst(pair.second->oldValues, values);
     }
   } else {
-    PRECICE_ERROR("Called extrapolation with order != 1,2!");
+    PRECICE_ASSERT(false, "Extrapolation order is invalid.");
```
Comment (BenjaminRueth) [inline src/cplscheme/BaseCouplingScheme.cpp:264]: Why `PRECICE_ASSERT(false, ...)` and not just `PRECICE_ERROR(...)`?

File: src/precice/impl/SolverInterfaceImpl.cpp [hunk 2]
```diff
@@ -1454,8 +1464,7 @@ const mesh::Mesh &SolverInterfaceImpl::mesh(const std::string &meshName) const
       return *context->mesh;
     }
   }
-  PRECICE_ERROR("Participant \"" << _accessorName
-                                 << "\" does not use mesh \"" << meshName << "\"!");
+  PRECICE_ASSERT(false, "Participant \"" << _accessorName << "\" does not use mesh \"" << meshName << "\".");
```
Comment (BenjaminRueth) [inline src/precice/impl/SolverInterfaceImpl.cpp:1467]: Why `PRECICE_ASSERT(false,...)` and not just `PRECICE_ERROR(...)`?

File: src/cplscheme/BaseCouplingScheme.cpp (hunk 1 above)
Comment (fsimonis) [reply src/cplscheme/BaseCouplingScheme.cpp:264]: This case should have been caught in the configuration.
It is thus an precice-internal problem and  should trigger an assertions.

File: src/precice/impl/SolverInterfaceImpl.cpp (hunk 2 above)
Comment (fsimonis) [reply src/precice/impl/SolverInterfaceImpl.cpp:1467]: This is only used in integration tests and means that the setup is incorrect. A user will never run into this code path.
Thus I would say that this should be an assertion.

**Pull request merged**